### PR TITLE
DM-4051 Split PageBuilder Publication date field into three separate inputs

### PR DESCRIPTION
**Pull request by camillevilla**
### JIRA issue link
[DM-4051](https://agile6.atlassian.net/browse/DM-4051) 

## Description - what does this code do?
This PR allows the PageBuilder Publication component to accept more date formats than `MM-DD-YYYY` by splitting date information into 3 separate fields. Form markup was modeled on the [USWDS Memorable Date component](https://designsystem.digital.gov/components/memorable-date/). 

## Testing done - how did you test it/steps on how can another person can test it 
1. Log in an as an admin and go to `/admin`. Create a new page group and page. 
2. Create a publication component with date: July 3, 2023
3. Create a publication component with date: July 2023
4. Go to the page. Confirm that both publications are rendering dates correctly and styled as cards.
5. Create a publication component with date: 2023
6. Go to the page. Confirm that all publications are rendering dates correctly and styled as a list.

## Screenshots, Gifs, Videos from application (if applicable)
Screenshot 2023-07-20 at 2 14 20 PM

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?type=design&node-id=6181%3A38985&mode=design&t=sBqG1SWXt5ECdM70-1

## Acceptance criteria
Publications can accept date formats such as: 
- July 3, 2023
- July 2023
- 2023

## Definition of done
- [ ] Unit tests written (if applicable)
- [x] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating JIRA issue
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/active_admin/resource/_page_publication_component_form.html.arb [hunk 1]
```diff
@@ -58,24 +58,57 @@ html = Arbre::Context.new do
...
+          label 'Month', for: "page_page_components_attributes_#{placeholder}_component_attributes_published_on_month", class: 'label'
+          select value: component&.published_on_month,
+                 id: "page_page_components_attributes_#{placeholder}_component_attributes_published_on_month",
+                 name: "page[page_components_attributes][#{placeholder}][component_attributes][published_on_month]" do
+                  option '', selected: component&.published_on_month === nil
+                  12.times do |i|
+                    # shorthand for generating: option '1 - January', value: 1, selected: component&.published_on_month === 1
+                    option "#{i + 1} - #{Date::MONTHNAMES[(i + 1)]}",
+                    value: i + 1,
+                    selected: component&.published_on_month === i + 1
```
Comment (PhilipDeFraties) [inline app/views/active_admin/resource/_page_publication_component_form.html.arb:73]:
```suggestion
                  1.upto(12) do |i|
                    # shorthand for generating: option '1 - January', value: 1, selected: component&.published_on_month === 1
                    option "#{i } - #{Date::MONTHNAMES[i ]}",
                    value: i,
                    selected: component&.published_on_month === i
```

Not important but we could do a nifty lil trick here ^

**Review by PhilipDeFraties — APPROVED** [review]
Looks good! I'm sure you are aware of this but there is no validation for the day field, so this is possible:

Screenshot 2023-07-20 at 6 04 03 PM

Also the form can be saved with month but no year, day but no month, but this is all stuff I would think can be added if we ever want to beef up the validations across the board seeing as many components have few.

Other than that and my suggestion, lgtm!

**Comment by camillevilla** [comment]
> Looks good! I'm sure you are aware of this but there is no validation for the day field, so this is possible:
> 
> Screenshot 2023-07-20 at 6 04 03 PM
> 
> Also the form can be saved with month but no year, day but no month, etc. Probably not important for this pr but just thought its worth mentioning, all stuff I would think can be added if we ever want to beef up the validations across the board for pb seeing as many components have few.*
> 
> Other than that and my suggestion, lgtm!

Ooh, good catch. Let me at least put a max on that...

**Review by PhilipDeFraties — APPROVED** [review]
Thanks for the update, glad you liked my suggestion, lgtm!

**Pull request merged**